inc/common/purge.php

```diff
@@ -651,8 +651,18 @@ function rocket_clean_cache_theme_update( $wp_upgrader, $hook_extra ) {
  * @param array $post_data Array of unslashed post data.
  */
 function rocket_clean_post_cache_on_slug_change( $post_id, $post_data ) {
-	if ( get_post_field( 'post_name', $post_id ) !== $post_data['post_name'] ) {
-        rocket_clean_files( get_the_permalink( $post_id ) );
-    }
+	// Bail out if the post status is draft, pending or auto-draft.
+	if ( in_array( get_post_field( 'post_status', $post_id ), [ 'draft', 'pending', 'auto-draft' ], true ) ) {
+		return;
+	}
+	// Bail out if the slug hasn't changed.
+	if ( get_post_field( 'post_name', $post_id ) === $post_data['post_name'] ) {
```

Comment:
@crystinutzaa `get_post_field()` can be an expensive operation as it goes through several different filters and processes. I'd suggest caching the post name in a local variable and then reusing it in both instance:

```php
$post_name = get_post_field( 'post_name', $post_id );

if ( $post_name === $post_data['post_name'] ) {
```

and same change on line 663.